packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts

```diff
@@ -120,6 +129,68 @@ export class InsuredBridgeL1Client {
     return this.whitelistedTokens[chainId];
   }
 
+  getRateModelForBlockNumber(l1Token: string, blockNumber: number | undefined = undefined): RateModel {
```

Comment:
I dont think this is how we want to implement this. IMHO we should add the functionality to pulling the rate model from the smart contracts (pretty much what you've got here) _into_ the SDK then use the SDK function in the bots. this way, the front end and anything else that also uses the SDK will use the same logic for pulling the rate model. if we do what you've got here we'll end up having duplicated code. 

(note that the implementation you've got here I think will do the trick, it's just a question of _where_ it's located).

Reply:
Will move this into the SDK after I finish up the implementation itself after you and matt's input

Reply:
Done!